### DISCUSSION TITLE
Purgecss Ignore width classes with breakpoints

### DISCUSSION THREAD

**Discussion by jillztom** [tailwindlabs/tailwindcss#1954] [Help]
I've some markups that I get through an API that contains tailwind width classes. 
```
eg: <div class="w-full lg:w-6/12>
```
When I compile the app of course purgecss is removing all unused classes like `lg:w-6/12`. 

Wondering what would be the best solution to avoid this issue? Whitelisting the classes using some regex? Does anyone have any samples for this regex or any other solutions?

**Comment by FalkoJoseph** [comment]
I’m actually having similar “issues” with PurgeCSS removing a lot of essential styling. I ended up not using it very often for production environments. Even though I absolutely love the idea behind the plugin, I hope they further clarify on some edge cases or issues.

**Comment by ivcreative** [comment]
This is kind of late but depending on what you are doing and how you compile either add the whitelist to the tailwind.config.js file or add it to your webpack or whatever you use.

If doing it from the tailwind file it should look kind of like this using whitelist in the example there is with the direct class or regex class:

```
purge: 
		{
		content: [],	
		// These options are passed through directly to PurgeCSS
		options: {
			whitelist: ['somvar', 'someother', 'lg\:w-6\/12'],
			whitelistPatterns: [/^var1-/, /^w-/, /^hover:/],
			whitelistPatternsChildren: [/^var2-/, /^var3-/,],
		}
	}
```

Now if you do it from your compiler I do it always from my webpack config file and use safelist combining both normal classes or with regex:

```
safelist: ['button', 'border', 'lg\:w-6\/12', '^lg:w-', ],
````

you can either use 'lg\:w-6\/12' or if you want to safelist or whitelist all the width because you also call them from the database then use the regex '^lg:w-'
